python/tests/conftest.py

```diff
@@ -120,6 +120,62 @@ def ts_fixture():
     return tables.tree_sequence()
 
 
+# 3 methods to return an example ts with internal samples, copied from test_highlevel.py
+@fixture(scope="session")
+def all_nodes_samples_example():
+    n = 5
+    ts = msprime.simulate(n, random_seed=10, mutation_rate=5)
+    assert ts.num_mutations > 0
+    tables = ts.dump_tables()
+    nodes = tables.nodes
+    flags = nodes.flags
+    # Set all nodes to be samples.
+    flags[:] = tskit.NODE_IS_SAMPLE
+    nodes.flags = flags
+    return tables.tree_sequence()
+
+
+@fixture(scope="session")
+def only_internal_samples_example():
+    n = 5
+    ts = msprime.simulate(n, random_seed=10, mutation_rate=5)
+    assert ts.num_mutations > 0
+    tables = ts.dump_tables()
+    nodes = tables.nodes
+    flags = nodes.flags
+    # Set just internal nodes to be samples.
+    flags[:] = 0
+    flags[n:] = tskit.NODE_IS_SAMPLE
+    nodes.flags = flags
+    return tables.tree_sequence()
+
+
+@fixture(scope="session")
+def mixed_node_samples_example():
+    n = 5
+    ts = msprime.simulate(n, random_seed=10, mutation_rate=5)
+    assert ts.num_mutations > 0
+    tables = ts.dump_tables()
+    nodes = tables.nodes
+    flags = nodes.flags
+    # Set a mixture of internal and leaf samples.
+    flags[:] = 0
+    flags[n // 2 : n + n // 2] = tskit.NODE_IS_SAMPLE
+    nodes.flags = flags
+    return tables.tree_sequence()
+
+
+@fixture(scope="session")
```

Comment:
Silently iterating over examples within a test is an anti-pattern now, we shouldn't do it for new code. We don't want a fixture that contains multiple examples, we want to parametrise the test with multiple examples.

Reply:
Oh yes, you mean something like this (I've been porting something similar into SLiM / pyslim, but using dicts, rather than a class)

```
class TreeSequenceExamples:
    """A simple container for example tree seqs - seems a bit unnatural though?"""
    @classmethod
    def example1(cls):
        ...
        return ts

    @classmethod
    def example2(cls):
        ...
        return ts

    @classmethod
    def list_examples_1_and_2(cls):
        return ["example1", "example2"]

    @classmethod
    def list_all_examples(cls):
        return [meth for meth in dir(cls) if meth.startswith "example"]


@pytest.fixture(scope="session", params=TreeSequenceExamples.list_all_examples())
def example_ts(request):
     return getattr(TreeSequenceExamples, request.param)(TreeSequenceExamples)

# In test file

@pytest.mark.parametrize('example_ts', argvalues=['example1', 'example2'])
def test_my_examples(example_ts):
    # do something with example_ts

# or equivalantly

@pytest.mark.parametrize('example_ts', argvalues=TreeSequenceExamples.list_examples_1_and_2())
def test_my_examples(example_ts):
    # do something with example_ts
```

Just a rough idea. You or @benjeffery can probably come up with a more elegant scheme?

Reply:
Oh, I guess rather than using a class, which seems a bit hacky, we could put all the examples into a file , and `import` that into `conftest.py`?

Reply:
In this particular case I would parameterise the fixture:
```python
n = 5
@fixture(scope="session",
         params=[('all', slice(None,None)), ('internal', slice(n,None)), ('mixed', slice(n // 2, n + n // 2))],
         ids=["all", "internal", "mixed"])
def node_samples_example(request):
    name, slice = request.param
    ts = msprime.simulate(n, random_seed=10, mutation_rate=5)
    assert ts.num_mutations > 0
    tables = ts.dump_tables()
    tables.metadata_schema = tskit.MetadataSchema({"codec":"json"})
    tables.metadata = {"name": name}
    nodes = tables.nodes
    flags = nodes.flags
    flags[:] = 0
    flags[slice] = tskit.NODE_IS_SAMPLE
    nodes.flags = flags
    return tables.tree_sequence()
```
The metadata is there so you can do `f"internal_sample_ts_{ts.metadata['name']}.svg"` in the test.

Reply:
Are these fixtures used elsewhere? Only widley used ones should be in `conftest.py`.

Reply:
I don't know, I'd rather not spend time investigating it right now. Can we just keep this to what we need for the change in question?

Reply:
I'm just sticking stuff into another (non urgent) fixture-related PR. For the moment I'll just revert the last commit in this PR and can add the extra tests when the other (fixture) PR eventually gets merged. No point working something out only to have to rip it out again, and these tests are only niceties anyway.